src/main/java/org/coffeetrain/doggifs/DogGifsApplication.kt

```diff
@@ -0,0 +1,17 @@
+package org.coffeetrain.doggifs
+
+import android.app.Application
+import android.content.Context
+
+class DogGifsApplication : Application() {
+  companion object {
+    private var dogGifRepository: DogGifRepository? = null
+    fun getDogGifRepository(context: Context): DogGifRepository {
+      // TODO add Dagger and cut this garbage out!
```

Comment:
maybe, but a simpler first step is to just expose it via `getSystemService` and have everybody access it via an extension method on Context.